### PR TITLE
[TH2-4864] Add features for filtration and response format for group search

### PR DESCRIPTION
**Pull request by OptimumCode**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt [hunk 1]
```diff
@@ -296,5 +323,6 @@ private fun StoredGroupedMessageBatch.toShortInfo(): String = "${group}:${firstM
 
 data class CradleGroupRequest(
     val sort: Boolean,
+    val include: Set<ProviderMessageStream> = emptySet(),
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt]: Why we can't execute all filtration via preFilter?

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt (hunk 1 above)
Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt]: We can. But it is used for filtration during pulling new messages. I didn't want to mix that logic with the new one (the prefilter is overwritten during pulling and we will have to somehow combine the filtration by streams with filtration required for pulling updates)

Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt]: @Nikita-Smirnov-Exactpro I can change it to be in preFilter if you need that

Comment (Nikita-Smirnov-Exactpro) [reply src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt]: Maybe split to two lambdas for each stage to use common approach for filtering?

Comment (OptimumCode) [reply src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt]: Done in next commits

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**